gammapy/maps/geom.py

```diff
@@ -640,6 +651,69 @@ def copy(self):
         return copy.deepcopy(self)
 
 
+    def group_table(self, edges):
```

Comment:
Is there a use case for other `MapAxis` than energy axis to perform this type of grouping? Should there be a specialized `EnergyAxis` for this?

Reply:
I was thinking maybe for the time binning, but there the logic will probably be different,  because time bins are probably represent by a single dataset each. I agree, if we introduce a specialised `EnergyAxis` class, this method should go there.